### PR TITLE
VEN-563 | BerthType on Berth mutations

### PR DESCRIPTION
**Pull request by a-rmz**
## Description :sparkles:
- Move most fields to common `BerthInput` class
- Change `create/update` mutations to accept either a `berth_type_id` or `width, length, berth_mooring_type` 
  - If the `berth_type_id` is passed, it uses that one
  - If dimensions are passed, it `get_or_create`s a suitable `BerthType`
  - If _both_ are passed, it raises an error
  - If only dimensions are passed but there's one missing, it raises an error

### Closes :no_good_woman:
**[VEN-563](https://helsinkisolutionoffice.atlassian.net/browse/VEN-563):** Do not expect BerthType id in create/update mutations for BerthNode

### Automated tests :gear:️
```shell
$ python resources/tests/test_resources_mutations.py
```

### Manual testing :construction_worker_man:
```graphql
mutation UPDATE_BERTH($input: UpdateBerthMutationInput!) {
  updateBerth(input: $input) {
    berth {
      id
      berthType {
        id
        width
        length
        mooringType
      }
    }
  }
}
```

```json
{
  "input": {
    "id":"QmVydGhOb2RlOjMzNDU5NTJhLTNiYzItNDE1NC1hOGJiLWQ5Yjg0NDlmMGI3Mg==",
    "mooringType": "STERN_BUOY_PLACE",
    "width": 6.66,
    "length": 3.3
  }
}
```

## Screenshots :camera_flash:
image

### CODE REVIEW

File: resources/schema.py [hunk 1]
```diff
@@ -454,26 +458,68 @@ def mutate_and_get_payload(cls, root, info, **input):
...
+        if berth_type_id:
+            berth_type = get_node_from_global_id(
+                info, berth_type_id, only_type=BerthTypeNode, nullable=False,
+            )
+        elif all(berth_type_params):
+            berth_type, created = BerthType.objects.get_or_create(
+                width=width, length=length, mooring_type=mooring_type
+            )
+        else:
+            raise VenepaikkaGraphQLError(_("Missing fields to associate a BerthType"))
```
Comment (igordavydsson) [inline resources/schema.py:480]: seems like this logic could be in a separate function 🤔 `berth_type = get_berth_type_from_berth_mutation_input(input)` or something...

File: resources/tests/test_resources_mutations.py [hunk 2]
```diff
@@ -229,8 +358,91 @@ def test_update_berth(berth, pier, berth_type, api_client):
...
+        },
+    }
+
+
+@pytest.mark.parametrize(
+    "api_client", ["harbor_services", "berth_services"], indirect=True,
+)
+def test_update_berth_existing_berth_type(berth, pier, berth_type, api_client):
+    global_id = to_global_id(BerthNode._meta.name, str(berth.id))
+    berth_type_id = to_global_id(BerthTypeNode._meta.name, str(berth_type.id))
```
Comment (igordavydsson) [inline resources/tests/test_resources_mutations.py:376]: you could maybe 
```python
assert not berth.berth_type == berth_type
```
🤔

Comment (a-rmz) [reply resources/tests/test_resources_mutations.py:376]: Good idea. 👍

File: resources/schema.py [hunk 3]
```diff
@@ -435,45 +435,77 @@ class AbstractBoatPlaceInput:
...
-        pier_id = graphene.ID()
-        berth_type_id = graphene.ID()
 
     berth = graphene.Field(BerthNode)
 
     @classmethod
     @change_permission_required(Berth)
     @transaction.atomic
     def mutate_and_get_payload(cls, root, info, **input):
+        # Check if the field should be updated
```
Comment (igordavydsson) [inline resources/schema.py:503]: nice 👍

Comment (a-rmz) [reply resources/schema.py:503]: :happy-cat:

**Pull request merged**